### ISSUE TITLE
image relighting demo videos

### ISSUE THREAD

**Issue by an2tong** [renjiaoyi/imagerelighting#3]
Thanks for sharing the great work!

I am trying to run the relighting code. However it seems both render_video.py and naive_insertion.py can not run directly due to missing some files there. Is it possible to share those files as well to make those two scripts (render_video.py and naive_insertion.py) to run as well? 

Also, regarding the monkey video showed in project page (https://renjiaoyi.github.io/relighting/), could you share the relighting code that uses the HDR image to render the money object as shown in the video? Thanks a lot!

**Comment by renjiaoyi** [comment]
The files are too many to share. For each HDR, we sample hundreds of views by rotating them. 

We highly suggest that you generate the SH coefficients of HDRs by simple least-squares fitting. 

A sample HDR [veranda_2k.hdr](https://github.com/renjiaoyi/imagerelighting/blob/main/Relighting/hdr/veranda_2k.hdr) and the coefficient  [veranda_2k.hdr.txt](https://github.com/renjiaoyi/imagerelighting/blob/main/Relighting/coeffs/veranda_2k.hdr.txt) are provided. You can use them to verify. 

The cutmaps are used as background for object insertion, we just crop the center part from HDRs. 

If needed, I can upload all sampled views for one HDR as references. 

Hope it helps. 

--the authors

**Comment by jwyndreee** [comment]
> We highly suggest that you generate the SH coefficients of HDRs by simple least-squares fitting.

Dear authors, sorry for my elementary question. I am new to the field and do not know how to generate SH coefficients from HDRs. Could you provide me any guidance? Sincere thanks.

**Comment by renjiaoyi** [comment]
> > We highly suggest that you generate the SH coefficients of HDRs by simple least-squares fitting.
> 
> Dear authors, sorry for my elementary question. I am new to the field and do not know how to generate SH coefficients from HDRs. Could you provide me any guidance? Sincere thanks.

Hi, I upload a Matlab file "getcoffs_new2.m" in folder "Relighting/coeffs/". It is used to compute SH coefficients from HDR files. In this file, HDR is rotated to generate coefficients for 360 degrees, used in video demos. 

Hope it helps. 

--the authors

**Comment by HavenFeng** [comment]
> > > We highly suggest that you generate the SH coefficients of HDRs by simple least-squares fitting.
> > 
> > 
> > Dear authors, sorry for my elementary question. I am new to the field and do not know how to generate SH coefficients from HDRs. Could you provide me any guidance? Sincere thanks.
> 
> Hi, I upload a Matlab file "getcoffs_new2.m" in folder "Relighting/coeffs/". It is used to compute SH coefficients from HDR files. In this file, HDR is rotated to generate coefficients for 360 degrees, used in video demos.
> 
> Hope it helps.
> 
> --the authors

Hi, thanks a lot for sharing the SH prefilter code, **but I think the img2 file is actually missing in the repo?** Could you upload that one as well? Thanks again for sharing this cool work!

**Comment by renjiaoyi** [comment]
> > > > We highly suggest that you generate the SH coefficients of HDRs by simple least-squares fitting.
> > > 
> > > 
> > > Dear authors, sorry for my elementary question. I am new to the field and do not know how to generate SH coefficients from HDRs. Could you provide me any guidance? Sincere thanks.
> > 
> > 
> > Hi, I upload a Matlab file "getcoffs_new2.m" in folder "Relighting/coeffs/". It is used to compute SH coefficients from HDR files. In this file, HDR is rotated to generate coefficients for 360 degrees, used in video demos.
> > Hope it helps.
> > --the authors
> 
> Hi, thanks a lot for sharing the SH prefilter code, **but I think the img2 file is actually missing in the repo?** Could you upload that one as well? Thanks again for sharing this cool work!

Yes, thanks for pointing out. I added img2.mat in the same folder.

**Comment by HavenFeng** [comment]
> > > > > We highly suggest that you generate the SH coefficients of HDRs by simple least-squares fitting.
> > > > 
> > > > 
> > > > Dear authors, sorry for my elementary question. I am new to the field and do not know how to generate SH coefficients from HDRs. Could you provide me any guidance? Sincere thanks.
> > > 
> > > 
> > > Hi, I upload a Matlab file "getcoffs_new2.m" in folder "Relighting/coeffs/". It is used to compute SH coefficients from HDR files. In this file, HDR is rotated to generate coefficients for 360 degrees, used in video demos.
> > > Hope it helps.
> > > --the authors
> > 
> > 
> > Hi, thanks a lot for sharing the SH prefilter code, **but I think the img2 file is actually missing in the repo?** Could you upload that one as well? Thanks again for sharing this cool work!
> 
> Yes, thanks for pointing out. I added img2.mat in the same folder.

Thanks a lot for adding it, now the extracted SH works well for the given tree sample!
